ldcomponents/data_system_configuration_builder.go

```diff
@@ -0,0 +1,199 @@
+	if d.endpoints.Streaming != "" {
+		streaming.BaseURI(d.endpoints.Streaming)
+	}
+	polling := PollingDataSourceV2()
+	if d.endpoints.Polling != "" {
+		polling.BaseURI(d.endpoints.Polling)
+	}
+	return d.Custom().Initializers(polling.AsInitializer()).Synchronizers(streaming, polling)
+}
+
+// Streaming configures the SDK to efficiently streams flag/segment data in the background,
+// allowing evaluations to operate on the latest data with no additional latency.
+func (d *DataSystemModes) Streaming() *DataSystemConfigurationBuilder {
+	streaming := StreamingDataSourceV2()
+	if d.endpoints.Streaming != "" {
+		streaming.BaseURI(d.endpoints.Streaming)
+	}
+	return d.Custom().Synchronizers(streaming, nil)
+}
+
+// Polling configures the SDK to regularly poll an endpoint for flag/segment data in the background.
+// This is less efficient than streaming, but may be necessary in some network environments.
+func (d *DataSystemModes) Polling() *DataSystemConfigurationBuilder {
+	polling := PollingDataSourceV2()
+	if d.endpoints.Polling != "" {
+		polling.BaseURI(d.endpoints.Polling)
+	}
+	return d.Custom().Synchronizers(polling, nil)
+}
+
+// Daemon configures the SDK to read from a persistent store integration that is populated by Relay Proxy
+// or other SDKs. The SDK will not connect to LaunchDarkly. In this mode, the SDK never writes to the data store.
+func (d *DataSystemModes) Daemon(store ss.ComponentConfigurer[ss.DataStore]) *DataSystemConfigurationBuilder {
+	return d.Custom().DataStore(store, ss.DataStoreModeRead)
+}
+
+// PersistentStore is similar to Default, with the addition of a
+// persistent store integration. Before data has arrived from LaunchDarkly, the SDK is able to
+// evaluate flags using data from the persistent store. Once fresh data is available, the SDK
+// will no longer read from the persistent store, although it will keep it up-to-date.
+func (d *DataSystemModes) PersistentStore(store ss.ComponentConfigurer[ss.DataStore]) *DataSystemConfigurationBuilder {
+	return d.Default().DataStore(store, ss.DataStoreModeReadWrite)
+}
+
+// Custom returns a builder suitable for creating a custom data acquisition strategy. You may configure
+// how the SDK uses a Persistent Store, how the SDK obtains an initial set of data, and how the SDK keeps data
+// up-to-date.
+func (d *DataSystemModes) Custom() *DataSystemConfigurationBuilder {
+	return &DataSystemConfigurationBuilder{}
+}
+
+// WithEndpoints configures the data system with custom endpoints for LaunchDarkly's streaming
+// and polling synchronizers. This method is not necessary for most use-cases, but can be useful for
+// testing or custom network configurations.
+//
+// Any endpoint that is not specified (empty string) will be treated as the default LaunchDarkly SaaS endpoint
+// for that service.
+func (d *DataSystemModes) WithEndpoints(endpoints Endpoints) *DataSystemModes {
+	if endpoints.Streaming != "" {
+		d.endpoints.Streaming = endpoints.Streaming
```

Comment:
If we wanted to somehow sanitize these endpoints, this might be the place to do it.